server/test/unit/server/webservices/views/test_repositories.py

```diff
@@ -0,0 +1,1977 @@
+
+
+class TestRepoResourceView(unittest.TestCase):
+    """
+    Tests for RepoResoureceView.
+    """
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_get_existing_repo(self, mock_factory, mock_resp):
+        """
+        Retrieve an existing repository.
+        """
+
+        mock_repo = {'mock_repo': 'somedata'}
+        mock_q_manager = mock_factory.repo_query_manager.return_value
+        mock_q_manager.find_by_id.return_value = mock_repo
+        mock_request = mock.MagicMock()
+        mock_request.GET = {}
+
+        repos_resource = RepoResourceView()
+        response = repos_resource.get(mock_request, 'mock_repo')
+
+        mock_resp.assert_called_once_with(mock_repo)
+        self.assertTrue(response is mock_resp.return_value)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_get_nonexisting_repo(self, mock_factory, mock_resp):
+        """
+        Retrieve a nonexisting repository.
+        """
+
+        mock_q_manager = mock_factory.repo_query_manager.return_value
+        mock_q_manager.find_by_id.return_value = None
+        mock_request = mock.MagicMock()
+
+        repos_resource = RepoResourceView()
+        try:
+            repos_resource.get(mock_request, 'mock_repo')
+        except pulp_exceptions.MissingResource, response:
+            pass
+        else:
+            raise AssertionError("MissingResource should be raised for a nonexisting repository")
+
+        self.assertEqual(response.http_status_code, 404)
+        self.assertTrue(response.error_code is error_codes.PLP0009)
+
+    @mock.patch('pulp.server.webservices.controllers.decorators._verify_auth',
+                new=assert_auth_READ())
+    @mock.patch(
+        'pulp.server.webservices.views.repositories.generate_json_response_with_pulp_encoder')
+    @mock.patch('pulp.server.webservices.views.repositories._merge_related_objects')
+    @mock.patch('pulp.server.webservices.views.repositories.manager_factory')
+    def test_get_existing_repo_with_details(self, mock_factory, mock_merge, mock_resp):
+        """
+        Retrieve an existing repository with details.
+        """
+
+        mock_repo = {'mock_repo': 'somedata'}
+        mock_q_manager = mock_factory.repo_query_manager.return_value
+        mock_q_manager.find_by_id.return_value = mock_repo
+        mock_request = mock.MagicMock()
+        mock_request.GET = {'details': 'true'}
+        mock_merge.side_effect = lambda x, y, z: z
```

Comment:
side_effect can be used to execute a function rather than returning the function, and it overrides return_value. In this case, I wanted `mock_merge(name, manager, repos)` to return repos.